test_set_expire fail on s390x

## Description

tests fail on IBM s390x.

## Steps to Reproduce

1. Package recipe https://build.opensuse.org/package/show/security:privacy/rnp
2. Build log https://build.opensuse.org/package/live_build_log/security:privacy/rnp/openSUSE_Factory_zSystems/s390x
3. Raw log https://build.opensuse.org/build/security:privacy/openSUSE_Factory_zSystems/s390x/rnp/_log

## Expected Behavior

Tests pass

## Actual Behavior

````
[ 4494s] test_set_dd (__main__.Keystore.test_set_expire) ... /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnpkeys/rnpkeys --homedir /tmp/rnpctmpk29az1jr/.rnp --password= --userid primary_with_empty_password@rnp --generate-key
[ 4494s] 
[ 4494s] Generating a new key...
[ 4494s] 
[ 4494s] sec   2048/RSA fd13a27e7c1bc7fd 2023-06-19 [SC] [EXPIRES 2025-06-18]
[ 4494s]       bece71aa50adcef92ca35e13fd13a27e7c1bc7fd
[ 4494s] uid           primary_with_empty_password@rnp
[ 4494s] ssb   2048/RSA f1db98fbaca6bce3 2023-06-19 [E] [EXPIRES 2025-06-18]
[ 4494s]       96ad7787221dbfa6e226e950f1db98fbaca6bce3
[ 4494s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnpkeys/rnpkeys --homedir /tmp/rnpctmpk29az1jr/.rnp --edit-key --set-expire -1 primary_with_empty_password@rnp
[ 4494s] Failed to set key expiration.
[ 4494s] 
[ 4494s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnp/rnp --list-packets /tmp/rnpctmpk29az1jr/.rnp/pubring.gpg
[ 4494s] 
[ 4494s] :off 0: packet header 0xc633 (tag 6, len 51)
[ 4494s] Public key packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1577369391 (Thu Dec 26 14:09:51 2019)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     public key material:
[ 4494s]         ecc p: 263 bits
[ 4494s]         ecc curve: Ed25519
[ 4494s]     keyid: 0x0451409669ffde3c
[ 4494s] :off 53: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 32 (Key revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172838 (Mon Jun 19 11:07:18 2023)
[ 4494s]         :type 29, len 1
[ 4494s]         reason for revocation: 0 (No reason)
[ 4494s]         message: 
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x81c6
[ 4494s]     signature material:
[ 4494s]         ecc r: 256 bits
[ 4494s]         ecc s: 254 bits
[ 4494s] :off 175: packet header 0xc27e (tag 2, len 126)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 32 (Key revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172840 (Mon Jun 19 11:07:20 2023)
[ 4494s]         :type 29, len 7
[ 4494s]         reason for revocation: 3 (Retired)
[ 4494s]         message: Custom
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xb2ec
[ 4494s]     signature material:
[ 4494s]         ecc r: 253 bits
[ 4494s]         ecc s: 256 bits
[ 4494s] :off 303: packet header 0xcd11 (tag 13, len 17)
[ 4494s] UserID packet
[ 4494s]     id: Alice <alice@rnp>
[ 4494s] :off 322: packet header 0xc290 (tag 2, len 144)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 19 (Positive User ID certification)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1577369391 (Thu Dec 26 14:09:51 2019)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x03 ( certify sign )
[ 4494s]         :type 11, len 4
[ 4494s]         preferred symmetric algorithms: AES-256, AES-192, AES-128, TripleDES (9, 8, 7, 2)
[ 4494s]         :type 21, len 5
[ 4494s]         preferred hash algorithms: SHA512, SHA384, SHA256, SHA224, SHA1 (10, 9, 8, 11, 2)
[ 4494s]         :type 22, len 3
[ 4494s]         preferred compression algorithms: ZLib, BZip2, ZIP (2, 3, 1)
[ 4494s]         :type 30, len 1
[ 4494s]         features: 0x01 ( mdc )
[ 4494s]         :type 23, len 1
[ 4494s]         key server preferences
[ 4494s]         no-modify: 1
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x249d
[ 4494s]     signature material:
[ 4494s]         ecc r: 255 bits
[ 4494s]         ecc s: 256 bits
[ 4494s] :off 468: packet header 0xce57 (tag 14, len 87)
[ 4494s] Public subkey packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1577455297 (Fri Dec 27 14:01:37 2019)
[ 4494s]     public key algorithm: 18 (ECDH)
[ 4494s]     public key material:
[ 4494s]         ecdh p: 515 bits
[ 4494s]         ecdh curve: brainpoolP256r1
[ 4494s]         ecdh hash algorithm: 8 (SHA256)
[ 4494s]         ecdh key wrap algorithm: 7
[ 4494s]     keyid: 0xdd23ceb7febeff17
[ 4494s] :off 557: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 24 (Subkey Binding Signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1577455297 (Fri Dec 27 14:01:37 2019)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x0c ( encrypt_comm encrypt_storage )
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xa1a0
[ 4494s]     signature material:
[ 4494s]         ecc r: 256 bits
[ 4494s]         ecc s: 254 bits
[ 4494s] :off 679: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 40 (Subkey revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172840 (Mon Jun 19 11:07:20 2023)
[ 4494s]         :type 29, len 1
[ 4494s]         reason for revocation: 0 (No reason)
[ 4494s]         message: 
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x8509
[ 4494s]     signature material:
[ 4494s]         ecc r: 253 bits
[ 4494s]         ecc s: 255 bits
[ 4494s] :off 801: packet header 0xc27d (tag 2, len 125)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 40 (Subkey revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172841 (Mon Jun 19 11:07:21 2023)
[ 4494s]         :type 29, len 6
[ 4494s]         reason for revocation: 2 (Compromised)
[ 4494s]         message: Other
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xf993
[ 4494s]     signature material:
[ 4494s]         ecc r: 255 bits
[ 4494s]         ecc s: 256 bits
[ 4494s] :off 928: packet header 0xc652 (tag 6, len 82)
[ 4494s] Public key packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1577369680 (Thu Dec 26 14:14:40 2019)
[ 4494s]     public key algorithm: 19 (ECDSA)
[ 4494s]     public key material:
[ 4494s]         ecc p: 515 bits
[ 4494s]         ecc curve: NIST P-256
[ 4494s]     keyid: 0x0b2b09f7d7ea6e0e
[ 4494s] :off 1012: packet header 0xcd11 (tag 13, len 17)
[ 4494s] UserID packet
[ 4494s]     id: Basil <basil@rnp>
[ 4494s] :off 1031: packet header 0xc290 (tag 2, len 144)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 19 (Positive User ID certification)
[ 4494s]     public key algorithm: 19 (ECDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0xd143811973b85e26b134b7a00b2b09f7d7ea6e0e (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1577369680 (Thu Dec 26 14:14:40 2019)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x03 ( certify sign )
[ 4494s]         :type 11, len 4
[ 4494s]         preferred symmetric algorithms: AES-256, AES-192, AES-128, TripleDES (9, 8, 7, 2)
[ 4494s]         :type 21, len 5
[ 4494s]         preferred hash algorithms: SHA512, SHA384, SHA256, SHA224, SHA1 (10, 9, 8, 11, 2)
[ 4494s]         :type 22, len 3
[ 4494s]         preferred compression algorithms: ZLib, BZip2, ZIP (2, 3, 1)
[ 4494s]         :type 30, len 1
[ 4494s]         features: 0x01 ( mdc )
[ 4494s]         :type 23, len 1
[ 4494s]         key server preferences
[ 4494s]         no-modify: 1
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0b2b09f7d7ea6e0e
[ 4494s]     lbits: 0x8387
[ 4494s]     signature material:
[ 4494s]         ecc r: 254 bits
[ 4494s]         ecc s: 255 bits
[ 4494s] :off 1177: packet header 0xc6c04d (tag 6, len 269)
[ 4494s] Public key packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1687172842 (Mon Jun 19 11:07:22 2023)
[ 4494s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4494s]     public key material:
[ 4494s]         rsa n: 2048 bits
[ 4494s]         rsa e: 17 bits
[ 4494s]     keyid: 0xfd13a27e7c1bc7fd
[ 4494s] :off 1449: packet header 0xcd1f (tag 13, len 31)
[ 4494s] UserID packet
[ 4494s]     id: primary_with_empty_password@rnp
[ 4494s] :off 1482: packet header 0xc2c08d (tag 2, len 333)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 19 (Positive User ID certification)
[ 4494s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0xbece71aa50adcef92ca35e13fd13a27e7c1bc7fd (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172848 (Mon Jun 19 11:07:28 2023)
[ 4494s]         :type 9, len 4
[ 4494s]         key expiration time: 63072000 seconds (730 days)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x03 ( certify sign )
[ 4494s]         :type 11, len 3
[ 4494s]         preferred symmetric algorithms: AES-256, AES-192, AES-128 (9, 8, 7)
[ 4494s]         :type 21, len 4
[ 4494s]         preferred hash algorithms: SHA256, SHA384, SHA512, SHA224 (8, 9, 10, 11)
[ 4494s]         :type 22, len 4
[ 4494s]         preferred compression algorithms: ZLib, BZip2, ZIP, Uncompressed (2, 3, 1, 0)
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0xfd13a27e7c1bc7fd
[ 4494s]     lbits: 0xc093
[ 4494s]     signature material:
[ 4494s]         rsa s: 2043 bits
[ 4494s] :off 1818: packet header 0xcec04d (tag 14, len 269)
[ 4494s] Public subkey packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1687172848 (Mon Jun 19 11:07:28 2023)
[ 4494s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4494s]     public key material:
[ 4494s]         rsa n: 2048 bits
[ 4494s]         rsa e: 17 bits
[ 4494s]     keyid: 0xf1db98fbaca6bce3
[ 4494s] :off 2090: packet header 0xc2c07c (tag 2, len 316)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 24 (Subkey Binding Signature)
[ 4494s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0xbece71aa50adcef92ca35e13fd13a27e7c1bc7fd (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172853 (Mon Jun 19 11:07:33 2023)
[ 4494s]         :type 9, len 4
[ 4494s]         key expiration time: 63072000 seconds (730 days)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x0c ( encrypt_comm encrypt_storage )
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0xfd13a27e7c1bc7fd
[ 4494s]     lbits: 0xa5c8
[ 4494s]     signature material:
[ 4494s]         rsa s: 2047 bits
[ 4494s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnpkeys/rnpkeys --homedir /tmp/rnpctmpk29az1jr/.rnp --edit-key --set-expire 0 wrongkey
[ 4494s] Secret keys matching 'wrongkey' not found.
[ 4494s] 
[ 4494s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnpkeys/rnpkeys --homedir /tmp/rnpctmpk29az1jr/.rnp --edit-key --set-expire 0 primary_with_empty_password@rnp
[ 4494s] 
[ 4494s] sec   2048/RSA fd13a27e7c1bc7fd 2023-06-19 [SC]
[ 4494s]       bece71aa50adcef92ca35e13fd13a27e7c1bc7fd
[ 4494s] uid           primary_with_empty_password@rnp
[ 4494s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnp/rnp --list-packets /tmp/rnpctmpk29az1jr/.rnp/pubring.gpg
[ 4494s] 
[ 4494s] :off 0: packet header 0xc633 (tag 6, len 51)
[ 4494s] Public key packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1577369391 (Thu Dec 26 14:09:51 2019)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     public key material:
[ 4494s]         ecc p: 263 bits
[ 4494s]         ecc curve: Ed25519
[ 4494s]     keyid: 0x0451409669ffde3c
[ 4494s] :off 53: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 32 (Key revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172838 (Mon Jun 19 11:07:18 2023)
[ 4494s]         :type 29, len 1
[ 4494s]         reason for revocation: 0 (No reason)
[ 4494s]         message: 
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x81c6
[ 4494s]     signature material:
[ 4494s]         ecc r: 256 bits
[ 4494s]         ecc s: 254 bits
[ 4494s] :off 175: packet header 0xc27e (tag 2, len 126)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 32 (Key revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172840 (Mon Jun 19 11:07:20 2023)
[ 4494s]         :type 29, len 7
[ 4494s]         reason for revocation: 3 (Retired)
[ 4494s]         message: Custom
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xb2ec
[ 4494s]     signature material:
[ 4494s]         ecc r: 253 bits
[ 4494s]         ecc s: 256 bits
[ 4494s] :off 303: packet header 0xcd11 (tag 13, len 17)
[ 4494s] UserID packet
[ 4494s]     id: Alice <alice@rnp>
[ 4494s] :off 322: packet header 0xc290 (tag 2, len 144)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 19 (Positive User ID certification)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1577369391 (Thu Dec 26 14:09:51 2019)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x03 ( certify sign )
[ 4494s]         :type 11, len 4
[ 4494s]         preferred symmetric algorithms: AES-256, AES-192, AES-128, TripleDES (9, 8, 7, 2)
[ 4494s]         :type 21, len 5
[ 4494s]         preferred hash algorithms: SHA512, SHA384, SHA256, SHA224, SHA1 (10, 9, 8, 11, 2)
[ 4494s]         :type 22, len 3
[ 4494s]         preferred compression algorithms: ZLib, BZip2, ZIP (2, 3, 1)
[ 4494s]         :type 30, len 1
[ 4494s]         features: 0x01 ( mdc )
[ 4494s]         :type 23, len 1
[ 4494s]         key server preferences
[ 4494s]         no-modify: 1
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x249d
[ 4494s]     signature material:
[ 4494s]         ecc r: 255 bits
[ 4494s]         ecc s: 256 bits
[ 4494s] :off 468: packet header 0xce57 (tag 14, len 87)
[ 4494s] Public subkey packet
[ 4494s]     version: 4
[ 4494s]     creation time: 1577455297 (Fri Dec 27 14:01:37 2019)
[ 4494s]     public key algorithm: 18 (ECDH)
[ 4494s]     public key material:
[ 4494s]         ecdh p: 515 bits
[ 4494s]         ecdh curve: brainpoolP256r1
[ 4494s]         ecdh hash algorithm: 8 (SHA256)
[ 4494s]         ecdh key wrap algorithm: 7
[ 4494s]     keyid: 0xdd23ceb7febeff17
[ 4494s] :off 557: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 24 (Subkey Binding Signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1577455297 (Fri Dec 27 14:01:37 2019)
[ 4494s]         :type 27, len 1
[ 4494s]         key flags: 0x0c ( encrypt_comm encrypt_storage )
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xa1a0
[ 4494s]     signature material:
[ 4494s]         ecc r: 256 bits
[ 4494s]         ecc s: 254 bits
[ 4494s] :off 679: packet header 0xc278 (tag 2, len 120)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 40 (Subkey revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172840 (Mon Jun 19 11:07:20 2023)
[ 4494s]         :type 29, len 1
[ 4494s]         reason for revocation: 0 (No reason)
[ 4494s]         message: 
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0x8509
[ 4494s]     signature material:
[ 4494s]         ecc r: 253 bits
[ 4494s]         ecc s: 255 bits
[ 4494s] :off 801: packet header 0xc27d (tag 2, len 125)
[ 4494s] Signature packet
[ 4494s]     version: 4
[ 4494s]     type: 40 (Subkey revocation signature)
[ 4494s]     public key algorithm: 22 (EdDSA)
[ 4494s]     hash algorithm: 8 (SHA256)
[ 4494s]     hashed subpackets:
[ 4494s]         :type 33, len 21
[ 4494s]         issuer fingerprint: 0x73edcc9119afc8e2dbbdcde50451409669ffde3c (20 bytes)
[ 4494s]         :type 2, len 4
[ 4494s]         signature creation time: 1687172841 (Mon Jun 19 11:07:21 2023)
[ 4494s]         :type 29, len 6
[ 4494s]         reason for revocation: 2 (Compromised)
[ 4494s]         message: Other
[ 4494s]     unhashed subpackets:
[ 4494s]         :type 16, len 8
[ 4494s]         issuer key ID: 0x0451409669ffde3c
[ 4494s]     lbits: 0xf993
[ 4494s]     signature material:
[ 4494s]         ecc r: 255 bits
[ 4495s]         ecc s: 256 bits
[ 4495s] :off 928: packet header 0xc652 (tag 6, len 82)
[ 4495s] Public key packet
[ 4495s]     version: 4
[ 4495s]     creation time: 1577369680 (Thu Dec 26 14:14:40 2019)
[ 4495s]     public key algorithm: 19 (ECDSA)
[ 4495s]     public key material:
[ 4495s]         ecc p: 515 bits
[ 4495s]         ecc curve: NIST P-256
[ 4495s]     keyid: 0x0b2b09f7d7ea6e0e
[ 4495s] :off 1012: packet header 0xcd11 (tag 13, len 17)
[ 4495s] UserID packet
[ 4495s]     id: Basil <basil@rnp>
[ 4495s] :off 1031: packet header 0xc290 (tag 2, len 144)
[ 4495s] Signature packet
[ 4495s]     version: 4
[ 4495s]     type: 19 (Positive User ID certification)
[ 4495s]     public key algorithm: 19 (ECDSA)
[ 4495s]     hash algorithm: 8 (SHA256)
[ 4495s]     hashed subpackets:
[ 4495s]         :type 33, len 21
[ 4495s]         issuer fingerprint: 0xd143811973b85e26b134b7a00b2b09f7d7ea6e0e (20 bytes)
[ 4495s]         :type 2, len 4
[ 4495s]         signature creation time: 1577369680 (Thu Dec 26 14:14:40 2019)
[ 4495s]         :type 27, len 1
[ 4495s]         key flags: 0x03 ( certify sign )
[ 4495s]         :type 11, len 4
[ 4495s]         preferred symmetric algorithms: AES-256, AES-192, AES-128, TripleDES (9, 8, 7, 2)
[ 4495s]         :type 21, len 5
[ 4495s]         preferred hash algorithms: SHA512, SHA384, SHA256, SHA224, SHA1 (10, 9, 8, 11, 2)
[ 4495s]         :type 22, len 3
[ 4495s]         preferred compression algorithms: ZLib, BZip2, ZIP (2, 3, 1)
[ 4495s]         :type 30, len 1
[ 4495s]         features: 0x01 ( mdc )
[ 4495s]         :type 23, len 1
[ 4495s]         key server preferences
[ 4495s]         no-modify: 1
[ 4495s]     unhashed subpackets:
[ 4495s]         :type 16, len 8
[ 4495s]         issuer key ID: 0x0b2b09f7d7ea6e0e
[ 4495s]     lbits: 0x8387
[ 4495s]     signature material:
[ 4495s]         ecc r: 254 bits
[ 4495s]         ecc s: 255 bits
[ 4495s] :off 1177: packet header 0xc6c04d (tag 6, len 269)
[ 4495s] Public key packet
[ 4495s]     version: 4
[ 4495s]     creation time: 1687172842 (Mon Jun 19 11:07:22 2023)
[ 4495s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4495s]     public key material:
[ 4495s]         rsa n: 2048 bits
[ 4495s]         rsa e: 17 bits
[ 4495s]     keyid: 0xfd13a27e7c1bc7fd
[ 4495s] :off 1449: packet header 0xcd1f (tag 13, len 31)
[ 4495s] UserID packet
[ 4495s]     id: primary_with_empty_password@rnp
[ 4495s] :off 1482: packet header 0xc2c087 (tag 2, len 327)
[ 4495s] Signature packet
[ 4495s]     version: 4
[ 4495s]     type: 19 (Positive User ID certification)
[ 4495s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4495s]     hash algorithm: 8 (SHA256)
[ 4495s]     hashed subpackets:
[ 4495s]         :type 33, len 21
[ 4495s]         issuer fingerprint: 0xbece71aa50adcef92ca35e13fd13a27e7c1bc7fd (20 bytes)
[ 4495s]         :type 2, len 4
[ 4495s]         signature creation time: 1687172853 (Mon Jun 19 11:07:33 2023)
[ 4495s]         :type 27, len 1
[ 4495s]         key flags: 0x03 ( certify sign )
[ 4495s]         :type 11, len 3
[ 4495s]         preferred symmetric algorithms: AES-256, AES-192, AES-128 (9, 8, 7)
[ 4495s]         :type 21, len 4
[ 4495s]         preferred hash algorithms: SHA256, SHA384, SHA512, SHA224 (8, 9, 10, 11)
[ 4495s]         :type 22, len 4
[ 4495s]         preferred compression algorithms: ZLib, BZip2, ZIP, Uncompressed (2, 3, 1, 0)
[ 4495s]     unhashed subpackets:
[ 4495s]         :type 16, len 8
[ 4495s]         issuer key ID: 0xfd13a27e7c1bc7fd
[ 4495s]     lbits: 0x2912
[ 4495s]     signature material:
[ 4495s]         rsa s: 2047 bits
[ 4495s] :off 1812: packet header 0xcec04d (tag 14, len 269)
[ 4495s] Public subkey packet
[ 4495s]     version: 4
[ 4495s]     creation time: 1687172848 (Mon Jun 19 11:07:28 2023)
[ 4495s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4495s]     public key material:
[ 4495s]         rsa n: 2048 bits
[ 4495s]         rsa e: 17 bits
[ 4495s]     keyid: 0xf1db98fbaca6bce3
[ 4495s] :off 2084: packet header 0xc2c07c (tag 2, len 316)
[ 4495s] Signature packet
[ 4495s]     version: 4
[ 4495s]     type: 24 (Subkey Binding Signature)
[ 4495s]     public key algorithm: 1 (RSA (Encrypt or Sign))
[ 4495s]     hash algorithm: 8 (SHA256)
[ 4495s]     hashed subpackets:
[ 4495s]         :type 33, len 21
[ 4495s]         issuer fingerprint: 0xbece71aa50adcef92ca35e13fd13a27e7c1bc7fd (20 bytes)
[ 4495s]         :type 2, len 4
[ 4495s]         signature creation time: 1687172853 (Mon Jun 19 11:07:33 2023)
[ 4495s]         :type 9, len 4
[ 4495s]         key expiration time: 63072000 seconds (730 days)
[ 4495s]         :type 27, len 1
[ 4495s]         key flags: 0x0c ( encrypt_comm encrypt_storage )
[ 4495s]     unhashed subpackets:
[ 4495s]         :type 16, len 8
[ 4495s]         issuer key ID: 0xfd13a27e7c1bc7fd
[ 4495s]     lbits: 0xa5c8
[ 4495s]     signature material:
[ 4495s]         rsa s: 2047 bits
[ 4495s] /home/abuild/rpmbuild/BUILD/rnp-v0.17.0/build/src/rnpkeys/rnpkeys --homedir /tmp/rnpctmpk29az1jr/.rnp --edit-key --set-expire 10 primary_with_empty_password@rnp
[ 4495s] 
[ 4495s] sec   2048/RSA fd13a27e7c1bc7fd 2023-06-19 [SC] [EXPIRED 2023-06-19]
[ 4495s]       bece71aa50adcef92ca35e13fd13a27e7c1bc7fd
[ 4495s] uid           primary_with_empty_password@rnp
[ 4495s] FAIL
[...]
[ 4495s] ======================================================================
[ 4495s] FAIL: test_set_expire (__main__.Keystore.test_set_expire)
[ 4495s] ----------------------------------------------------------------------
[ 4495s] Traceback (most recent call last):
[ 4495s]   File "/home/abuild/rpmbuild/BUILD/rnp-v0.17.0/src/tests/cli_tests.py", line 2003, in test_set_expire
[ 4495s]     self.assertRegex(out, r'(?s)^.*\[EXPIRES .*')
[ 4495s] AssertionError: Regex didn't match: '(?s)^.*\\[EXPIRES .*' not found in '\nsec   2048/RSA fd13a27e7c1bc7fd 2023-06-19 [SC] [EXPIRED 2023-06-19]\n      bece71aa50adcef92ca35e13fd13a27e7c1bc7fd\nuid           primary_with_empty_password@rnp\n'
[ 4495s] 
````

--- a/src/tests/cli_tests.py
+++ b/src/tests/cli_tests.py
@@ -1997,14 +1997,14 @@ class Keystore(unittest.TestCase):
         matches = re.findall(r'(key expiration time: 63072000 seconds \(730 days\))', out)
         self.assertEqual(len(matches), 1)
 
-        # Expires in 10 seconds
-        ret, out, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--edit-key', '--set-expire', '10', 'primary_with_empty_password@rnp'])
+        # Expires in 60 seconds
+        ret, out, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--edit-key', '--set-expire', '60', 'primary_with_empty_password@rnp'])
         self.assertEqual(ret, 0)
         self.assertRegex(out, r'(?s)^.*\[EXPIRES .*')
 
         ret, out, _ = run_proc(RNP, ['--list-packets', kpath])
         self.assertEqual(ret, 0)
-        self.assertRegex(out, r'(?s)^.*key expiration time: 10 seconds \(0 days\).*')
+        self.assertRegex(out, r'(?s)^.*key expiration time: 60 seconds \(0 days\).*')
 
         # Expires in 10 hours
         ret, out, _ = run_proc(RNPK, ['--homedir', RNPDIR, '--edit-key', '--set-expire', '10h', 'primary_with_empty_password@rnp'])

